torch/distributed/distributed_c10d.py

```diff
@@ -1752,10 +1775,9 @@ def _new_process_group_helper(
             "created, please use a different group name"
         )
 
-    if device_id is not None and (device_id.index is None or device_id.type != "cuda"):
+    if device_id is not None and device_id.index is None:
```

Comment:
I think it's a common problem for non-CUDA device, then should we separate this change in other PR?